alerter.c

```diff
@@ -1,32 +1,53 @@
 #include <stdio.h>
 #include <assert.h>
 
+#define Test_Code 1
+
 int alertFailureCount = 0;
 
+#ifdef Test_Code
 int networkAlertStub(float celcius) {
+    int returnValue = 0;
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
     // Return 200 for ok
     // Return 500 for not-ok
     // stub always succeeds and returns 200
-    return 200;
+    if(celcius > 500.0)
+    {
+        returnValue = 500;
+    }
+    else
+    {
+        returnValue = 200;
+    }
+    return returnValue;
 }
+#endif
 
 void alertInCelcius(float farenheit) {
+    int returnCode = 0;
     float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
+    #ifdef Test_Code

```

Comment:
what is the alternative for #ifdef ?
stub is used only in test environments or development environments. It is used when we do not have the production code that can be integrated. In such cases, it should be abstracted by using interfaces or delegates.
Dependency injection concept can be applied here.